src/pack_unpack.c

```diff
@@ -348,6 +348,33 @@ static json_t *pack_string(scanner_t *s, va_list *ap)
     }
 }
 
+static json_t *pack_object_inter(scanner_t *s, va_list *ap, int need_incref)
+{
+    int nullable;
+    json_t *json;
+
+    next_token(s);
+    nullable = token(s) == '?';
```

Comment:
Instead of `int nullable` could you create a var `char ntoken` and set it to `token(s)`?  This way you can check for `ntoken == '?'` or `ntoken == '*'` below.  I'd prefer this over reading `s->next_token.token` directly.  I know other code reads `s->next_token.token` directly but I feel we should move away from this as it's only set by calls to `prev_token()`, not calls to `next_token()`.

Reply:
Fixed. Thanks!